complete/src/integration-test/groovy/intranet/backend/AnnouncementControllerSpec.groovy

```diff
@@ -58,39 +63,41 @@ class AnnouncementControllerSpec extends Specification {
 
     def "test detail of an announcement contains body in both version 1.0 and 2.0"() {
         given:
-        RestBuilder rest = new RestBuilder()
+        def announcementId = 2 as Long
+        HttpClient client = HttpClient.create("http://localhost:${serverPort}/".toURL())
+        HttpRequest request = HttpRequest.GET(UriBuilder.of("/announcements/${announcementId}").build())
+                .header("Accept-Version", "1.0");
 
         when: 'Requesting announcements for version 1.0'
-        def annoucementId = 2 as Long
-        def resp = rest.get("http://localhost:${serverPort}/announcements/${annoucementId}") {
-            header("Accept-Version", "1.0")
-        }
+        HttpResponse<String> resp = client.toBlocking().exchange(request, String)
+        JSONObject jsonObject = new JSONObject(resp.body())
```

Comment:
You can bind to a `Map` no need to bind to string and then create a `JSONObject`